File: app/src/main/java/com/prox/voicechanger/adapter/FileVideoAdapter.java
package com.prox.voicechanger.adapter;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.content.Context;
import android.graphics.BitmapFactory;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.prox.voicechanger.databinding.DialogOptionVideoBinding;
import com.prox.voicechanger.databinding.DialogPlayVideoBinding;
import com.prox.voicechanger.databinding.ItemFileVideoBinding;
import com.prox.voicechanger.model.FileVoice;
import com.prox.voicechanger.ui.dialog.OptionVideoDialog;
import com.prox.voicechanger.ui.dialog.PlayVideoDialog;
import com.prox.voicechanger.utils.NumberUtils;
import com.prox.voicechanger.viewmodel.FileVoiceViewModel;

import java.io.File;
import java.util.ArrayList;
import java.util.List;

public class FileVideoAdapter extends RecyclerView.Adapter<FileVideoAdapter.FileVideoViewHolder> {
    private List<FileVoice> fileVideos;
    private final Context context;
    private final Activity activity;
    private final FileVoiceViewModel model;

    public FileVideoAdapter(Context context, Activity activity, FileVoiceViewModel model) {
        this.context = context;
        this.activity = activity;
        this.model = model;
    }

    @SuppressLint("NotifyDataSetChanged")
    public void setFileVideos(List<FileVoice> fileVideos) {
        if (fileVideos != null) {
            this.fileVideos = fileVideos;
        }else {
            this.fileVideos = new ArrayList<>();
        }
        notifyDataSetChanged();
    }

    public List<FileVoice> getFileVideos() {
        if (fileVideos == null) {
            return new ArrayList<>();
        } else {
            return fileVideos;
        }
    }

    @NonNull
    @Override
    public FileVideoViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemFileVideoBinding binding = ItemFileVideoBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
        return new FileVideoViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull FileVideoViewHolder holder, int position) {
        FileVoice fileVideo = fileVideos.get(position);
        if (new File(fileVideo.getImageVideo()).exists()){
            holder.binding.imgFile.setImageBitmap(BitmapFactory.decodeFile(fileVideo.getImageVideo()));
        }
        holder.binding.txtNameFile.setText(fileVideo.getName());
        holder.binding.txtSize.setText(NumberUtils.formatAsTime(fileVideo.getDuration()) + " | " + NumberUtils.formatAsSize(fileVideo.getSize()));
        holder.binding.txtDate.setText(NumberUtils.formatAsDate(fileVideo.getDate()));
        holder.binding.btnOption.setOnClickListener(view -> {
            OptionVideoDialog dialog = new OptionVideoDialog(
                    context,
                    activity,
                    DialogOptionVideoBinding.inflate(activity.getLayoutInflater()),
                    model,
                    fileVideo);
            dialog.show();
        });
        holder.binding.btnPlay.setOnClickListener(view -> {
            PlayVideoDialog dialog = new PlayVideoDialog(
                    context,
                    DialogPlayVideoBinding.inflate(activity.getLayoutInflater()),
                    fileVideo.getPath());
            dialog.show();
        });
    }

    @Override
    public int getItemCount() {
        if (fileVideos == null) {
            return 0;
        }
        return fileVideos.size();
    }

    public static class FileVideoViewHolder extends RecyclerView.ViewHolder {
        private final ItemFileVideoBinding binding;

        public FileVideoViewHolder(@NonNull ItemFileVideoBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }
}

File: app/src/main/java/com/prox/voicechanger/adapter/FileVideoAdapter.kt
package com.prox.voicechanger.adapter

import com.prox.voicechanger.utils.ConvertersUtils.toBitmap
import com.prox.voicechanger.utils.NumberUtils.formatAsTime
import com.prox.voicechanger.utils.NumberUtils.formatAsSize
import com.prox.voicechanger.utils.NumberUtils.formatAsDate
import androidx.recyclerview.widget.RecyclerView
import android.annotation.SuppressLint
import android.view.ViewGroup
import android.view.LayoutInflater
import android.app.Activity
import android.content.Context
import com.prox.voicechanger.viewmodel.FileVoiceViewModel
import com.prox.voicechanger.adapter.FileVideoAdapter.FileVideoViewHolder
import com.prox.voicechanger.ui.dialog.OptionVideoDialog
import com.prox.voicechanger.ui.dialog.PlayVideoDialog
import android.view.View
import com.prox.voicechanger.databinding.DialogOptionVideoBinding
import com.prox.voicechanger.databinding.DialogPlayVideoBinding
import com.prox.voicechanger.databinding.ItemFileVideoBinding
import com.prox.voicechanger.model.FileVoice
import java.util.ArrayList

class FileVideoAdapter(
    private val context: Context,
    private val activity: Activity,
    private val model: FileVoiceViewModel
) : RecyclerView.Adapter<FileVideoViewHolder>() {
    private var fileVideos: List<FileVoice?>? = null
    @SuppressLint("NotifyDataSetChanged")
    fun setFileVideos(fileVideos: List<FileVoice?>?) {
        if (fileVideos != null) {
            this.fileVideos = fileVideos
        } else {
            this.fileVideos = ArrayList()
        }
        notifyDataSetChanged()
    }

    fun getFileVideos(): List<FileVoice?> {
        return fileVideos ?: ArrayList()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FileVideoViewHolder {
        val binding =
            ItemFileVideoBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return FileVideoViewHolder(binding)
    }

    override fun onBindViewHolder(holder: FileVideoViewHolder, position: Int) {
        val fileVideo = fileVideos!![position]
        holder.binding.imgFile.setImageBitmap(toBitmap(fileVideo!!.image))
        holder.binding.txtNameFile.text = fileVideo.name
        holder.binding.txtSize.text =
            formatAsTime(fileVideo.duration) + " | " + formatAsSize(fileVideo.size)
        holder.binding.txtDate.text = formatAsDate(fileVideo.date)
        holder.binding.btnOption.setOnClickListener {
            val dialog = OptionVideoDialog(
                context,
                activity,
                DialogOptionVideoBinding.inflate(activity.layoutInflater),
                model,
                fileVideo
            )
            dialog.show()
        }
        holder.binding.btnPlay.setOnClickListener {
            val dialog = PlayVideoDialog(
                context,
                DialogPlayVideoBinding.inflate(activity.layoutInflater),
                fileVideo.path
            )
            dialog.show()
        }
    }

    override fun getItemCount(): Int {
        return if (fileVideos == null) {
            0
        } else fileVideos!!.size
    }

    class FileVideoViewHolder(val binding: ItemFileVideoBinding) : RecyclerView.ViewHolder(
        binding.root
    )
}